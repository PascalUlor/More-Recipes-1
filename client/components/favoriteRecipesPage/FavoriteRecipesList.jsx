import React from 'react';
import PropTypes from 'prop-types';
import FavoriteRecipe from './FavoriteRecipe.jsx';


/**
 * @description displays a of list favorite recipe or message of no recipes
 * @method FavoriteRecipesList
 * 
 * @param { object } favorites - object of a single favorite recipe
 * @param { function } setCurrentRecipe - sets currently selected recipe
 * 
 * @returns { jsx } jsx - renders FavoriteRecipeList component
 */
const FavoriteRecipesList = ({ favorites, setCurrentRecipe }) => {
  if (favorites.length === 0) {
    /**
     * @description displays no favorite recipes message
     * 
     * @returns { jsx } jsx - renders no favorite recipes message
     */
    return (
      <div className="not-found lead p-3 mt-3 mt-5">
        <i className=
        "fa fa-exclamation-triangle fa-3x pb-3 text-warning d-block"></i>
        You have no favorite recipes
      </div>
    );
  }

  const favoriteRecipes = favorites
    .map((favorite, index) => (index <= 5) &&
      <FavoriteRecipe
        key={favorite.id}
        favorite={favorite}
        setCurrentRecipe={setCurrentRecipe}/>);

  /**
   * @description displays a list favorite recipe
   * 
   * @returns { jsx } jsx - renders list of favorite recipes component
   */
  return (
    <div className="card-deck mb-4 mt-4">
      <div className="row w-100">
        { favoriteRecipes }
      </div>
    </div>
  );
};

FavoriteRecipesList.propTypes = {
  favorites: PropTypes.array.isRequired,
  setCurrentRecipe: PropTypes.func.isRequired
};

export default FavoriteRecipesList;
