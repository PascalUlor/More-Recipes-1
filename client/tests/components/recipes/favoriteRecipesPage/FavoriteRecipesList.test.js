/* eslint-disable */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import FavoriteRecipesList
  from '../../../../components/favoriteRecipesPage/FavoriteRecipesList.jsx';

const Recipe = {
  recipeImage: 'recipeImage.jpg',
  User: { fullName: '' }
};
const props = {
  setCurrentRecipe: jest.fn(),
  favorites: [{
    id: 1,
    Recipe
  }, {
    id: 2,
    Recipe
  }]
};

describe('FavoriteRecipesList component', () => {
  it('should render correctly', () => {
    const mountWrapper = mount((
      <Router>
        <FavoriteRecipesList {...props}/>
      </Router>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const newProps = {
      ...props,
      favorites: []
    };
    const mountWrapper = mount((
      <Router>
        <FavoriteRecipesList {...newProps}/>
      </Router>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });
});
