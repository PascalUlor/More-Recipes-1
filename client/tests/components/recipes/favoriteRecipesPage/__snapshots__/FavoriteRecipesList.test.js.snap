// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteRecipesList component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FavoriteRecipesList
      favorites={
        Array [
          Object {
            "Recipe": Object {
              "User": Object {
                "fullName": "",
              },
              "recipeImage": "recipeImage.jpg",
            },
            "id": 1,
          },
          Object {
            "Recipe": Object {
              "User": Object {
                "fullName": "",
              },
              "recipeImage": "recipeImage.jpg",
            },
            "id": 2,
          },
        ]
      }
      setCurrentRecipe={[MockFunction]}
    >
      <div
        className="card-deck mb-4 mt-4"
      >
        <div
          className="row w-100"
        >
          <FavoriteRecipe
            favorite={
              Object {
                "Recipe": Object {
                  "User": Object {
                    "fullName": "",
                  },
                  "recipeImage": "recipeImage.jpg",
                },
                "id": 1,
              }
            }
            key="1"
            setCurrentRecipe={[MockFunction]}
          >
            <div
              className="col-12 col-sm-12 col-md-6 col-lg-4 pt-2"
            >
              <div
                className="card"
              >
                <Link
                  replace={false}
                  to="/recipes/undefined/recipe-details"
                >
                  <a
                    href="/recipes/undefined/recipe-details"
                    onClick={[Function]}
                  >
                    <img
                      alt="Card image cap"
                      className="card-img-top img-fluid"
                      src="recipeImage.jpg"
                    />
                  </a>
                </Link>
                <div
                  className="card-body p-2"
                >
                  <h5
                    className="card-title mt-2"
                  />
                  <p
                    className="text-muted w-100"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user text-info"
                    />
                  </p>
                  <div>
                    <Link
                      className="btn btn-sm btn-outline-info mr-2"
                      replace={false}
                      to="/recipes/undefined/recipe-details"
                    >
                      <a
                        className="btn btn-sm btn-outline-info mr-2"
                        href="/recipes/undefined/recipe-details"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-eye"
                        />
                         View
                      </a>
                    </Link>
                    <button
                      className="btn btn-sm btn-outline-danger ml-2"
                      data-target="#deleteRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="fa fa-trash"
                      />
                       Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </FavoriteRecipe>
          <FavoriteRecipe
            favorite={
              Object {
                "Recipe": Object {
                  "User": Object {
                    "fullName": "",
                  },
                  "recipeImage": "recipeImage.jpg",
                },
                "id": 2,
              }
            }
            key="2"
            setCurrentRecipe={[MockFunction]}
          >
            <div
              className="col-12 col-sm-12 col-md-6 col-lg-4 pt-2"
            >
              <div
                className="card"
              >
                <Link
                  replace={false}
                  to="/recipes/undefined/recipe-details"
                >
                  <a
                    href="/recipes/undefined/recipe-details"
                    onClick={[Function]}
                  >
                    <img
                      alt="Card image cap"
                      className="card-img-top img-fluid"
                      src="recipeImage.jpg"
                    />
                  </a>
                </Link>
                <div
                  className="card-body p-2"
                >
                  <h5
                    className="card-title mt-2"
                  />
                  <p
                    className="text-muted w-100"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user text-info"
                    />
                  </p>
                  <div>
                    <Link
                      className="btn btn-sm btn-outline-info mr-2"
                      replace={false}
                      to="/recipes/undefined/recipe-details"
                    >
                      <a
                        className="btn btn-sm btn-outline-info mr-2"
                        href="/recipes/undefined/recipe-details"
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-eye"
                        />
                         View
                      </a>
                    </Link>
                    <button
                      className="btn btn-sm btn-outline-danger ml-2"
                      data-target="#deleteRecipeModal"
                      data-toggle="modal"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="fa fa-trash"
                      />
                       Delete
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </FavoriteRecipe>
        </div>
      </div>
    </FavoriteRecipesList>
  </Router>
</BrowserRouter>
`;

exports[`FavoriteRecipesList component should render correctly 2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FavoriteRecipesList
      favorites={Array []}
      setCurrentRecipe={[MockFunction]}
    >
      <div
        className="not-found lead p-3 mt-3 mt-5"
      >
        <i
          className="fa fa-exclamation-triangle fa-3x pb-3 text-warning d-block"
        />
        You have no favorite recipes
      </div>
    </FavoriteRecipesList>
  </Router>
</BrowserRouter>
`;
